### PR TITLE
Remove mocha-eslint dep

### PR DESCRIPTION
**Pull request by hknustwmf**
Removed mocha-eslint dep and removed eslint-config-node-services dep and updated other packages to latest versions and provided overrides for MW lint settings where needed.

https://phabricator.wikimedia.org/T210460

### CODE REVIEW

File: lib/kafka_factory.js [hunk 1]
```diff
@@ -121,23 +121,23 @@ class MetadataWatch extends EventEmitter {
...
-                        this.emit('topics_changed', topics);
-                    }
-                    this._knownTopics = topics;
-                })
-                .catch(e => this.emit('error', e));
-                // TODO: make configurable
-            }, 10000);
-        })
-        .thenReturn(this);
+            .then((topics) => {
```
Comment (d00rman) [inline lib/kafka_factory.js]: Noooo. Please, please, let's not do this. Indenting the `'then` blocks makes the code way less readable.

File: lib/kafka_factory.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ class GuaranteedProducer extends kafka.Producer {
 class AutopollingProducer extends kafka.Producer {
     constructor(config, topicConfig, logger) {
         super(config, topicConfig);
-        this.on('event.error', e => logger.log('error/producer', {
+        this.on('event.error', (e) => logger.log('error/producer', {
```
Comment (d00rman) [inline lib/kafka_factory.js]: Hm, I'd like to add an exception for this, these parentheses here seem really unnecessary.

File: lib/kafka_factory.js [hunk 3]
```diff
@@ -200,10 +200,10 @@ class KafkaFactory {
         }
 
         this._consumerConf = Object.assign({}, CONSUMER_DEFAULTS, kafkaConf.consumer || {});
-        this._consumerConf['metadata.broker.list'] = kafkaConf.metadata_broker_list;
+        this._consumerConf[ 'metadata.broker.list' ] = kafkaConf.metadata_broker_list;
 
         this._producerConf = Object.assign({}, PRODUCER_DEFAULTS, kafkaConf.producer || {});
-        this._producerConf['metadata.broker.list'] = kafkaConf.metadata_broker_list;
+        this._producerConf[ 'metadata.broker.list' ] = kafkaConf.metadata_broker_list;
```
Comment (d00rman) [inline lib/kafka_factory.js]: Spaces? Oh please no.

File: lib/rule.js [hunk 4]
```diff
@@ -1,18 +1,16 @@
...
 const stringify = require('json-stable-stringify');
 const HyperSwitch = require('hyperswitch');
 const Sampler = require('./sampler');
 const Template = HyperSwitch.Template;
 
-const DEFAULT_RETRY_DELAY = 60000;  // One minute minimum retry delay
-const DEFAULT_RETRY_FACTOR = 6;     // Exponential back-off
-const DEFAULT_RETRY_LIMIT = 2;      // At most two retries
+const DEFAULT_RETRY_DELAY = 60000; // One minute minimum retry delay
+const DEFAULT_RETRY_FACTOR = 6; // Exponential back-off
```
Comment (d00rman) [inline lib/rule.js]: Heh, ironically this makes comments less visible.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "change-propagation",
-  "version": "0.9.0",
+  "version": "0.9.1",
```
Comment (d00rman) [inline package.json]: Let's not touch the version nor production-level deps in a PR about code style, please.

File: test/index.js [hunk 6]
```diff
@@ -3,10 +3,4 @@
 
 // Run jshint as part of normal testing
 require('mocha-jshint')();
```
Comment (Pchelolo) [inline test/index.js]: This has been deprecated long time ago. You can remove this and get rid of the dependency.

File: test/feature/update_rules.js [hunk 7]
```diff
@@ -559,6 +559,7 @@ describe('update rules', function() {
                 return validate(body[0]);
             })
             .reply(200, {});
+            console.log("Here")
```
Comment (Pchelolo) [inline test/feature/update_rules.js]: Ups

**Comment by Pchelolo** [comment]
The tests are failing most likely because of the updated dependencies.

**Comment by Pchelolo** [comment]
Let's not do any meaningful code changes (even in tests) as a part of a style PR. I would prefer you reverting all the dependencies/devDeps changes not related to style.

File: lib/kafka_factory.js (hunk 1 above)
Comment (hknustwmf) [reply lib/kafka_factory.js]: I didn't like it either but I did not want to make too many exceptions. Will remove.

File: lib/kafka_factory.js (hunk 2 above)
Comment (hknustwmf) [reply lib/kafka_factory.js]: Sure. This one I for one prefer over no parentheses but I can remove it.

File: lib/kafka_factory.js (hunk 3 above)
Comment (hknustwmf) [reply lib/kafka_factory.js]: I cringed too :). Will remove again

File: lib/rule.js (hunk 4 above)
Comment (hknustwmf) [reply lib/rule.js]: So undo or run with it?

File: package.json (hunk 5 above)
Comment (hknustwmf) [reply package.json]: ok. going back

File: test/index.js (hunk 6 above)
Comment (hknustwmf) [reply test/index.js]: Ok

**Comment by hknustwmf** [comment]
I am going to close this PR and start over. Should be a lot faster than undoing half the changes.

**Pull request closed without merging**